File: wimf-processor/src/Post/Core/Domain/Post.cs
﻿using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;
using Utils;

namespace PostApi
{
    public class Post: BaseEntity
    {
        
        public const int TitleMinLength = 3;
        public const int TitleMaxLength = 40;
        public const int ContentMaxLength = 600;

#nullable disable
        [BsonElement("postId")] public string PostId { get; set; }

        [BsonElement("postType")] public string PostType { get; set; }

        [BsonElement("title")] public string Title { get; set; }

        [BsonElement("content")] public string Content { get; set; }

        [BsonElement("location")] public UserLocation Location { get; set; }

        [BsonElement("userId")] public string UserId { get; set; }
        
        [BsonElement("createdAt")] public long CreatedAt { get; set; }

#nullable enable        
        [BsonElement("userInfo")] public UserInfo? UserInfo { get; set; }
        
        [BsonElement("attachements")] public IEnumerable<string>? Attachements { get; set; }
        
        [BsonElement("updatedAt")] public long? UpdatedAt { get; set; }
        
    }
}

File: wimf-site/src/core/domain/Post/Post.ts
import { Attachment } from './Attachment';
import { WimfLocation } from '../WimfLocation';
import { UserInfo } from '../UserInfo';

export type PostType = 'Found' | 'Lost' | 'Spotted';

export interface Post {
  readonly postId: string;
  readonly postType: PostType;
  readonly title: string;
  readonly content: string;
  readonly location: WimfLocation;
  readonly personId: string;
  readonly createdAt: number;
  readonly userInfo?: UserInfo;
  readonly attachments?: Attachment[];
  readonly reward?: number;
  readonly updatedAt?: number;
}

export interface CreatePostPayload {
  readonly postType: PostType;
  readonly title: string;
  readonly content: string;
  readonly location: WimfLocation;
  readonly userId: string;
  readonly userInfo?: UserInfo;
  readonly attachements?: string[];
  readonly reward?: number;
}
